guides/basics/introduction-to-avalonia-concepts.md

```diff
@@ -0,0 +1,3686 @@
+
+**Note**: the so called XAML namespace URLs do **not** have to refer to any valid URL that really exists on a web and the computer does **not** have to be online in order for them to work. 
+
+The example showing various ways to define custom XAML namespace is located under [NP.Demos.XamlNamespacesSample.sln](https://github.com/npolyak/NP.Avalonia.Demos/tree/main/NP.Demos.XamlSamples/NP.Demos.XamlNamespacesSample) solution. Download its code from github and open the solution in Visual Studio \(or Rider\). 
+
+You can see that the solution consists of 3 projects: the main project NP.Demos.XamlNamespacesSample and two projects on which the main project depends: Dependency1Proj and Dependency2Proj:
+
+![](../../.gitbook/assets/sample1.png)
+
+Compile and run the solution - here is what you are going to see:
+
+![](../../.gitbook/assets/sample2.png)
+
+There are 5 square buttons of different colors stacked vertically within a window. 
+
+Here is the relevant code of MainWindow.xaml file:
+
+```markup
+<Window xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:dep1="clr-namespace:Dependency1Proj;assembly=Dependency1Proj"
+        xmlns:dep1_sub_Folder="clr-namespace:Dependency1Proj.SubFolder;assembly=Dependency1Proj"
+        xmlns:local="clr-namespace:NP.Demos.XamlNamespacesSample"
+        xmlns:dep2="https://avaloniademos.com/xaml"
+        ...>
+  <StackPanel HorizontalAlignment="Center"
+              VerticalAlignment="Center">
+    <dep1:RedButton/>
+    <dep1_sub_Folder:BlueButton/>
+    <local:BrownButton/>
+    <dep2:GreenButton/>
+    <dep2:GrayButton/>
+  </StackPanel>
+</Window>
+
+```
+
+There are four custom namespaces defined by the following lines of the top XML Tag:
+
+```markup
+xmlns:dep1="clr-namespace:Dependency1Proj;assembly=Dependency1Proj"
+xmlns:dep1_test_Folder="clr-namespace:Dependency1Proj.SubFolder;assembly=Dependency1Proj"
+xmlns:local="clr-namespace:NP.Demos.XamlNamespacesSample"
+xmlns:dep2="https://avaloniademos.com/xaml"
+```
+
+Different buttons are are referred two by the corresponding XAML namespace prefixes. Let us take a look at `<RedButton/>` `RedButton` class is defined as a C\# class under Dependency1Proj project. Here is its code:
+
+```csharp
+...
+namespace Dependency1Proj
+{
+    public class RedButton : Button, IStyleable
+    {
+        Type IStyleable.StyleKey => typeof(Button);
+
+        public RedButton()
+        {
+            Background = new SolidColorBrush(Colors.Red);
+            Width = 30;
+            Height = 30;
+        }
+    }
+}
+```
+
+The line
+
+```csharp
+Type IStyleable.StyleKey => typeof(Button);
+```
```

Comment:
Those samples can be misleading for developers who are new to Avalonia and don't know styling system basics yet. They may think that its the only way to make button with custom width\height etc